oxidation/libparsec/crates/client_connection/src/error.rs

```diff
@@ -1,46 +1,45 @@
 // Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 (eventually AGPL-3.0) 2016-present Scille SAS
 
-use std::{error::Error, fmt::Display};
+use thiserror::Error;
 
-pub type Result<T> = core::result::Result<T, CommandError>;
+pub type CommandResult<T> = core::result::Result<T, CommandError>;
 
 /// Sending a command isn't risk-free, we have multiple possible way to fail.
 /// Also note we only deal with *transport* related errors here (i.e. *deserialization* / *http* / *tcp* related stuff),
 /// hence dealing with the `status` field of the response message is left to the caller
-#[derive(Debug)]
+#[derive(Debug, Error)]
 pub enum CommandError {
-    /// We failed to retrieve the reply.
-    NoResponse(reqwest::Error),
+    /// Any invalid content
+    #[error("Invalid content")]
+    BadContent,
+
+    /// The organization has expired
+    #[error("Trial organization has expired")]
+    ExpiredOrganization,
+
     /// We receive a response but with an unexpected status code.
+    #[error("unexpected response status {0}")]
     InvalidResponseStatus(reqwest::StatusCode, reqwest::Response),
+
     /// We failed to deserialize the reply.
+    #[error("failed to deserialize the response: {0}")]
     InvalidResponseContent(libparsec_protocol::DecodeError),
-}
 
-impl Error for CommandError {
-    fn source(&self) -> Option<&(dyn Error + 'static)> {
-        match self {
-            CommandError::InvalidResponseContent(e) => Some(e),
-            CommandError::NoResponse(e) => Some(e),
-            _ => None,
-        }
-    }
-}
+    /// We failed to retrieve the reply.
+    #[error("failed to retrieving the response: {0}")]
+    NoResponse(reqwest::Error),
 
-impl Display for CommandError {
-    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        match self {
-            CommandError::NoResponse(reason) => {
-                write!(f, "failed to retrieving the response: {reason}")
-            }
-            CommandError::InvalidResponseStatus(code, _response) => {
-                write!(f, "unexpected response status {code}")
-            }
-            CommandError::InvalidResponseContent(reason) => {
-                write!(f, "failed to deserialize the response: {reason}")
-            }
-        }
-    }
+    /// The user has beed revoked
+    #[error("Device has been revoked")]
+    RevokedUser,
+
+    /// Failed to serialize the request
+    #[error("{0}")]
+    Serialization(libparsec_protocol::EncodeError),
+
+    /// The version is not supported
+    #[error("Unsupported API version: {0}")]
+    UnsupportedApiVersion(String),
```

Comment:
The backend should reply with the list of supported api versions, Can you add those ?

Reply:
According to the `rpc`
```Python
def _rpc_msgpack_rep(data: dict[str, object], api_version: ApiVersion) -> Response:
    return Response(
        response=packb(data),
        # Unlike REST, RPC doesn't use status to encode operational result
        status=200,
        content_type=CONTENT_TYPE_MSGPACK,
        headers={"Api-Version": str(api_version)},
    )
    ```
    There is only one :thinking:

Reply:
That response I when the api version is compatible between the client/server, 
it's not the message sent when the api are incompatble

Reply:
Done, I missed the header Supported-Api-Versions